### ISSUE TITLE
Rebuild watchdog

### ISSUE THREAD

**Issue by dreikanter** [dreikanter/public-static#65]
- `pub run -w` should make enable watchdog mode. In this mode any source directory update triggers site rebuild.
- Watchdog should run the build process synchronously (next rebuild could be started only after the current one is completed).
- It would be nice to have Windows support.

Some relevant libraries:
- http://pythonhosted.org/watchdog/
- http://timgolden.me.uk/python/win32_how_do_i/watch_directory_for_changes.html
- http://pyinotify.sourceforge.net/
